loader/checkpoint.go

```diff
@@ -346,6 +381,16 @@ func (cp *RemoteCheckPoint) GenSQL(filename string, offset int64) string {
 	return sql
 }
 
+// UpdateOffset implements CheckPoint.UpdateOffset
+func (cp *RemoteCheckPoint) UpdateOffset(filename string, offset int64) {
+	fields := strings.Split(filename, ".")
+	if len(fields) != 2 && len(fields) != 3 {
+		cp.logCtx.L().Error("can't get db and table from filename in checkpoint UpdateOffset", zap.String("filename", filename))
+	}
+	db, table := fields[0], fields[1]
+	cp.restoringFiles[db][table][filename][0] = offset
```

Comment:
No mutex needed because we only modify `cp.restoringFiles[db][table][filename]` in one goroutine?

Reply:
structure of map is initiated in `(*RemoteCheckPoint).Init`, thus it won't cause rehash and allocation, won't access invalid memory. same job of same SQL file are processed in one goroutine